File: library/src/main/java/com/ebr163/inspector/rule/EmailRule.java
package com.ebr163.inspector.rule;

import android.util.Patterns;

/**
 * Created by Bakht
 * on 09.01.2018.
 */

public class EmailRule extends AbstractRule<String> {

    public EmailRule(String errorMessage) {
        super(errorMessage);
    }

    @Override
    public boolean verify(String value) {
        return Patterns.EMAIL_ADDRESS.matcher(value).matches();
    }
}

File: library/src/main/java/com/ebr163/inspector/rule/EmailRule.kt
package com.ebr163.inspector.rule

import android.util.Patterns

/**
 * Created by Bakht
 * on 09.01.2018.
 */

class EmailRule(errorMessage: String) : AbstractRule<String>(errorMessage) {

    override fun verify(value: String?): Boolean = Patterns.EMAIL_ADDRESS.matcher(value).matches()
}
